### DISCUSSION TITLE
Automations on SOC

### DISCUSSION THREAD

**Discussion by JamesWJPS** [StealthChesnut/HA-FoxESS-Modbus#39] [Q&A] [answered]
Hi,

I would like to setup some automations based on Battery Soc (i.e. turn on and off smart plugs). I have create an automation and set the condition to a numeric state of the battery between 90 and 100. However when i try to test this is states:

"In 'numeric_state': In 'numeric_state' condition: entity sensor.battery_soc state 'measurement' cannot be processed as a number"

Is this becuase the soc is measured with a % sign at the end. How can i get round this?

James

**Comment by JamesWJPS** [comment, marked as answer]
I have worked it out. Sorry my lack of knowledge with HA. Within a condition i was trying to look at an Attiribute not realising if that was left blank it was the numeric state.

**Reply by StealthChesnut** [reply]
All good James! Would you mind adding a reply with a screenshot in case it helps someone else running up on the same problem please?

**Comment by JamesWJPS** [comment]
So I was trying to add Battery SoC as a condition on an automation. Although there is the drop down for attribute you can leave this blank, like this:

![image](https://user-images.githubusercontent.com/23116673/184358654-23da9490-7764-471a-815d-f72deeb0ef24.png)
